testsuites/syncgateway/upgrade/test_upgrade.py

```diff
@@ -0,0 +1,480 @@
+import os
+import random
+import time
+
+from keywords.couchbaseserver import verify_server_version
+from libraries.testkit.cluster import Cluster
+from keywords.utils import log_info, host_for_url
+from keywords.SyncGateway import (verify_sg_accel_version,
+                                  verify_sync_gateway_version,
+                                  verify_sg_accel_product_info,
+                                  verify_sync_gateway_product_info,
+                                  SyncGateway)
+from keywords.ClusterKeywords import ClusterKeywords
+from keywords.MobileRestClient import MobileRestClient
+from keywords import attachment
+from couchbase.bucket import Bucket
+from keywords.constants import SDK_TIMEOUT
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from requests.exceptions import HTTPError
+
+
+def test_upgrade(params_from_base_test_setup):
+    """ The initial versions of SG and CBS has already been provisioned at this point
+        We have to upgrade them to the upgraded versions
+    """
+    cluster_config = params_from_base_test_setup['cluster_config']
+    mode = params_from_base_test_setup['mode']
+    xattrs_enabled = params_from_base_test_setup['xattrs_enabled']
+    ls_url = params_from_base_test_setup["ls_url"]
+    server_version = params_from_base_test_setup['server_version']
+    sync_gateway_version = params_from_base_test_setup['sync_gateway_version']
+    server_upgraded_version = params_from_base_test_setup['server_upgraded_version']
+    sync_gateway_upgraded_version = params_from_base_test_setup['sync_gateway_upgraded_version']
+    sg_url = params_from_base_test_setup['sg_url']
+    sg_admin_url = params_from_base_test_setup['sg_admin_url']
+    num_docs = int(params_from_base_test_setup['num_docs'])
+    cbs_platform = params_from_base_test_setup['cbs_platform']
+    cbs_toy_build = params_from_base_test_setup['cbs_toy_build']
+    sg_conf = "{}/resources/sync_gateway_configs/sync_gateway_default_functional_tests_{}.json".format(os.getcwd(), mode)
+
+    # Add data to liteserv
+    client = MobileRestClient()
+    log_info("ls_url: {}".format(ls_url))
+    ls_db = client.create_database(ls_url, name="ls_db")
+
+    # Create user and session on SG
+    sg_user_channels = ["sg_user_channel"]
+    sg_db = "db"
+    sg_user_name = "sg_user"
+    sg_user_password = "password"
+    client.create_user(
+        url=sg_admin_url,
+        db=sg_db,
+        name=sg_user_name,
+        password=sg_user_password,
+        channels=sg_user_channels
+    )
+    sg_session = client.create_session(url=sg_admin_url, db=sg_db, name=sg_user_name, password=sg_user_password)
+
+    # Start continuous push pull replication ls_db_one <-> sg_db_one
+    log_info("Starting replication from liteserv to sync gateway")
```

Comment:
In my opinion, if you put log_info, you can remove comment , it is kind of duplicate, we can use one for same purpose

Reply:
Removed